### DISCUSSION TITLE
[BUG] The request has an invalid grant parameter : code

### DISCUSSION THREAD

**Discussion by Zzzoux** [amzn/selling-partner-api-models#3064] [General]
When using the website authorization workflow, I found that the spapi obtained through step 3. Amazon sends you the authorization information_ oauth_ Code to request

Step 4. Your application exchanges the LWA authorization code for a LWA refresh token

"error_description": "The request has an invalid grant parameter : code",
"error": "invalid_grant"

I'm sure my code time does not exceed the expiration time of 5 minutes, but it still happens. How can I modify it to get the token correctly

Here are my request parameters:
![image](https://user-images.githubusercontent.com/112464076/187365565-0eca9807-d2f0-4999-a6c6-0512dc660806.png)
![image](https://user-images.githubusercontent.com/112464076/187365914-3fade8cc-b9e6-47e3-8f53-4a9918e5582e.png)

**Comment by sungolivia** [comment]
hard to say exactly what it is but maybe it would help if you can share some screenshots of how you got the authorization_code? Below blog mentions adding a header and it could be that as well.

check out this [blog](https://developer-docs.amazon.com/sp-api-blog/docs/migrate-seller-authorizations-from-mws-to-sp-api#step-4-exchange-authorization-code-for-an-lwa-refresh-token) for step by step with Postman screenshots!

**Comment by arorasahil1209** [comment]
have you found solution for this ?

**Comment by Zzzoux** [comment]
Confirm your client_ ID and client_ Whether secret matches APP


&nbsp;


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "amzn/selling-partner-api-docs"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年12月3日(星期六) 凌晨4:30
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [amzn/selling-partner-api-docs] [BUG] The request has an invalid grant parameter : code (Discussion amzn/selling-partner-api-docs#2590)





 
have you found solution for this ?

**Comment by kishanK111** [comment]
Generated code in oauth is single time use. Generate code again with oauth flow and try again.

**Comment by manav-kasare** [comment]
@Zzzoux
I am facing the same issue, have you manged to fix it?

**Comment by Zzzoux** [comment]
Confirm your client_ ID and client_ Whether secret matches APP


&nbsp;
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "amzn/selling-partner-api-docs"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年5月23日(星期二) 凌晨0:06
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [amzn/selling-partner-api-docs] [BUG] The request has an invalid grant parameter : code (Discussion amzn/selling-partner-api-docs#2590)





 
@Zzzoux
 I am facing the same issue, have you manged to fix it?

**Comment by Jordanlagan** [comment]
Not sure if it's the same issue you guys are facing but I figured out that you cannot create your own LWA application to use with SP-API. 

When you create an app client in Developer Central, it automatically creates a set of LWA credentials and it appears that these are the only LWA credentials you may use. 

I was getting:
"error_description": "The request has an invalid grant parameter : code",
"error": "invalid_grant"

Everything was correct except I guess my client_id and secret didn't match up to what Amazon expected. Switched that over and it works.
